src/controller/controller/wingsail/controllers.py

```diff
@@ -0,0 +1,78 @@
+import math
+
+from controller.common.lut import LUT
+
+
+class WingsailController:
+    """
+    The controller class for computing trim tab angles for controlling the mainsail.
+
+    Args:
+    - chord_width_main_sail (float): The chord width of the main sail.
+    - kinematic_viscosity (float): The kinematic viscosity of the fluid.
+    - lut (LUT): A lookup table containing Reynolds numbers and corresponding desired angles of
+      attack.
+    """
+
+    def __init__(self, chord_width_main_sail: float, kinematic_viscosity: float, lut: LUT):
+        self.chord_width_main_sail = chord_width_main_sail
+        self.kinematic_viscosity = kinematic_viscosity
+        self.lut: LUT = lut
+
+    def _compute_reynolds_number(self, apparent_wind_speed: float) -> float:
+        """
+        Computes the Reynolds number for the main sail.
+
+        Args:
+        - apparent_wind_speed (float): The apparent wind speed in meters per second.
+
+        Returns:
+        - reynolds_number (float): The computed Reynolds number for the main sail.
+        """
+        reynolds_number: float = (
+            apparent_wind_speed * self.chord_width_main_sail
+        ) / self.kinematic_viscosity
+        return reynolds_number
+
+    def _compute_trim_tab_angle(
+        self, reynolds_number: float, apparent_wind_direction: float
+    ) -> float:
+        """
+        Computes the trim tab angle based on Reynolds number and apparent wind direction.
+
+        Args:
+        - reynolds_number (float): The Reynolds number.
+        - apparent_wind_direction (float): The absolute bearing (true heading) of the wind.
+        Degrees, 0° means the apparent wind is blowing from the bow to the stern of the boat,
+        increase CW
+        Range: -180 < direction <= 180 for symmetry
+
+        Returns:
+        - trim_tab_angle (float): The computed trim tab angle based on the provided
+        Reynolds number and apparent wind direction.
```

Comment:
Clarify your assumptions about the controller.

You actually are not considering $|\alpha| > 90\degree$, but this is not necessarily obvious both in the documentation or the code.

Also, I interpreted absolute bearing as 0 degrees North, increasing CW, but it sounds like you may be describing something different.

Based on our discussion:
- 0 Degrees, from bow to stern, and increasing CCW
![image](https://github.com/UBCSailbot/sailbot_workspace/assets/69879126/b25b1f1a-addb-42a9-bed5-50a8303934b7)

Reply:
Updated it
```
 def _compute_trim_tab_angle(self, reynolds_number: float, apparent_wind_direction: float) -> float:
        Computes the trim tab angle based on Reynolds number and apparent wind direction. During
        this computation, it is assumed that the wingsail is aligned in the direction of the wind,
        with the rear of the wingsail (where the trim tab is located) pointing in the same
        direction as the wind is blowing.

        Args:
        - reynolds_number (float): The Reynolds number.
        - apparent_wind_direction (float): direction of the wind, in degrees. Here 0° means the
        apparent wind is blowing from the bow to the stern of the boat, and the angle increases CCW.
        Range: -180 < direction <= 180 for symmetry

        Returns:
        - trim_tab_angle (float): The computed trim tab angle based on the provided
        Reynolds number and apparent wind direction.
```